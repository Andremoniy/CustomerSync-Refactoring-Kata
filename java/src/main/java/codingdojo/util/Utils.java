package codingdojo.util;

import codingdojo.misc.Customer;
import codingdojo.misc.CustomerDataLayer;
import codingdojo.match.CustomerMatches;
import codingdojo.misc.ShoppingList;

public class Utils {
    public static void subLoadCompanyCustomer(CustomerMatches matches, Customer matchByExternalId, Customer matchByMasterId, Customer matchByCompanyNumber) {
        if (matchByExternalId != null) {
            matches.setCustomer(matchByExternalId);
            matches.setMatchTerm("ExternalId");
            if (matchByMasterId != null) matches.addDuplicate(matchByMasterId);
        } else {
            if (matchByCompanyNumber != null) {
                matches.setCustomer(matchByCompanyNumber);
                matches.setMatchTerm("CompanyNumber");
            }
        }
    }

    public static void updateShoppingList(Customer customer, ShoppingList consumerShoppingList, CustomerDataLayer customerDataLayer1) {
        customer.addShoppingList(consumerShoppingList);
        customerDataLayer1.updateShoppingList(consumerShoppingList);
        customerDataLayer1.upd(customer);
    }
}
