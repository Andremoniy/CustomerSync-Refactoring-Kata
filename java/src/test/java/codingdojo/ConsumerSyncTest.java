package codingdojo;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.util.Arrays;

import static org.junit.jupiter.api.Assertions.assertFalse;


public class ConsumerSyncTest {

    @Test
    public void testSyncByExternalId(){
        String externalId = "12345";

        Consumer consumer = new Consumer();
        consumer.setExternalId(externalId);
        consumer.setName("Standard Consumer");
        consumer.setAddress(new Address("123 main st", "Helsingborg", "SE-123 45"));
        consumer.setCompanyNumber("470813-8895");
        consumer.setPreferredStore("Nordstan");
        consumer.setShoppingLists(Arrays.asList(new ShoppingList("lipstick", "blusher")));

        Customer customer = new Customer();
        customer.setExternalId(externalId);
        customer.setCompanyNumber(consumer.getCompanyNumber());
        customer.setCustomerType(CustomerType.COMPANY);
        customer.setInternalId("45435");

        FakeDatabase db = new FakeDatabase();
        db.addCustomer(customer);
        ConsumerSync sut = new ConsumerSync(db);

        StringBuilder toAssert = new StringBuilder();
        toAssert.append("BEFORE:\n");
        toAssert.append(db.printContents());

        toAssert.append("\nSYNCING THIS:\n");
        toAssert.append(ConsumerPrinter.print(consumer, ""));

        // ACT
        boolean created = sut.syncWithDataLayer(consumer);

        assertFalse(created);
        toAssert.append("\nAFTER:\n");
        toAssert.append(db.printContents());
        Approvals.verify(toAssert);
    }

    private Consumer standardConsumer() {
        Consumer consumer = new Consumer();
        consumer.setExternalId("12345");
        consumer.setName("Standard Consumer");
        consumer.setAddress(new Address("123 main st", "Helsingborg", "SE-123 45"));
        consumer.setCompanyNumber("470813-8895");
        consumer.setPreferredStore("Nordstan");
        consumer.setShoppingLists(Arrays.asList(new ShoppingList("lipstick", "blusher")));
        return consumer;
    }
}
