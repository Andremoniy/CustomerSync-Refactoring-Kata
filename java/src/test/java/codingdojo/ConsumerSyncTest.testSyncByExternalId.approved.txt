BEFORE:
Fake Database.
All Customers {

    Customer {
        externalId='12345'
        masterExternalId='null'
        companyNumber='470813-8895'
        internalId='45435'
        name='null'
        customerType=COMPANY
        preferredStore='null'
        address='null'
        shoppingLists=[]
    }

}
All Shopping Lists{
[]
}
SYNCING THIS:
Consumer {
    externalId='12345'
    companyNumber='470813-8895'
    name='Standard Consumer'
    preferredStore='Nordstan'
    address='123 main st, SE-123 45 Helsingborg'
    shoppingLists=[[lipstick, blusher]]}
AFTER:
Fake Database.
All Customers {

    Customer {
        externalId='12345'
        masterExternalId='null'
        companyNumber='470813-8895'
        internalId='45435'
        name='Standard Consumer'
        customerType=COMPANY
        preferredStore='Nordstan'
        address='123 main st, SE-123 45 Helsingborg'
        shoppingLists=[[lipstick, blusher]]
    }

}
All Shopping Lists{
[[lipstick, blusher]]
}